### PR TITLE
Exams by Kseniia Horai

### PR DESCRIPTION
**Pull request by KseniiaHorai**
1) validate.js and sum.js (with commits)
2) section.js, sample.js, reverse.js (fixed without commits)

### CODE REVIEW

**Review by notnuff — APPROVED** [review]
All fine!

File: Solutions/reverse.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+'use strict';
+
+const reverseObject = (object) => {
+    const keys = Object.keys(object);
+    for (let item of keys) {
+        const newKey = object[item];
+        object[newKey] = item;
+        delete object[item];
+    }
+    return object;
```
Comment (tshemsedinov) [inline Solutions/reverse.js:11]: Try not to change incoming parameters but generate new object

File: Solutions/reverse.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Reverse dict, exchange keys and values
+'use strict';
+
+const reverseObject = (object) => {
+    const keys = Object.keys(object);
+    for (let item of keys) {
+        const newKey = object[item];
+        object[newKey] = item;
+        delete object[item];
```
Comment (tshemsedinov) [inline Solutions/reverse.js:9]: Try to avoid `delete`

File: Solutions/section.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// Split string into array by the first occurrence of separator
+'use strict';
+
+const sectionString = (string, separator) => {
+    let i = string.indexOf(separator);
+     if (i === -1 || separator === '') {
+       return [string, ''];
+     } else {
```
Comment (tshemsedinov) [inline Solutions/section.js:8]: We do not need else here

File: Tasks/validate.js [hunk 4]
```diff
@@ -1,24 +1,19 @@
...
-  if (!T.includes(' ')) return false;
-  {
-    for (C of T) {
-      if (C === ' ') continue;
-      if (
-        C.toLowerCase().charCodeAt(0) >= 97 &&
-        C.toLowerCase().charCodeAt(0) <= 122
-      ) {
-      } else {
+const isValid = (name) => {
```
Comment (tshemsedinov) [inline Tasks/validate.js:4]: Naming hint: better name `validate` or `check` (verbs)

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
You need linter and or fix code alignment manually

File: Tasks/sum.js [hunk 5]
```diff
@@ -1,20 +1,23 @@
...
+  if (!Array.isArray(array)) {
+    return 'It is not an array';
+  }
+
+  if (array.length === 0) {
+    return 0;
+  }
+  
+  for (let item of array) {
+    if (typeof(item) === 'number') {
```
[the remark this answers is not in the document]
Comment (KseniiaHorai) [reply Tasks/sum.js:15]: Fixed here:
https://github.com/KseniiaHorai/Exams/blob/master/Tasks/sum.js

File: Tasks/validate.js [hunk 6]
```diff
@@ -1,24 +1,19 @@
...
+    if (typeof(name) !== 'string' || name.trim() === '' || !name.includes(' ')) {
         return false;
+    }
+  
+    for (let letter of name) {
+      if (letter === ' ') continue;
+      if ( letter.toLowerCase().charCodeAt(0) < 97 || letter.toLowerCase().charCodeAt(0) > 122) {
```
[the remark this answers is not in the document]
Comment (KseniiaHorai) [reply Tasks/validate.js:11]: Fixed code here:
https://github.com/KseniiaHorai/Exams/blob/master/Tasks/validate.js

**Pull request closed without merging**